pkg/controllers/workapplier/availability_tracker.go

```diff
@@ -247,6 +253,30 @@ func trackPDBAvailability(curObj *unstructured.Unstructured) (ManifestProcessing
 	return ManifestProcessingAvailabilityResultTypeNotYetAvailable, nil
 }
 
+// trackServiceExportAvailability tracks the availability of a service export in the member cluster.
+// It is available if it has a weight annotation and the ServiceExportValid condition is True.
+func trackServiceExportAvailability(curObj *unstructured.Unstructured) (ManifestProcessingAvailabilityResultType, error) {
+	var svcExport fleetnetworkingv1alpha1.ServiceExport
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(curObj.Object, &svcExport); err != nil {
+		return ManifestProcessingAvailabilityResultTypeFailed, controller.NewUnexpectedBehaviorError(err)
+	}
+
+	weight, hasAnnotation := svcExport.Annotations[objectmeta.ServiceExportAnnotationWeight]
+	validCond := meta.FindStatusCondition(svcExport.Status.Conditions, string(fleetnetworkingv1alpha1.ServiceExportValid))
+	conflictCond := meta.FindStatusCondition(svcExport.Status.Conditions, string(fleetnetworkingv1alpha1.ServiceExportConflict))
+
+	isValid := validCond != nil && validCond.Status == metav1.ConditionTrue
+	hasNoConflict := conflictCond != nil && conflictCond.Status == metav1.ConditionFalse
+
+	if hasAnnotation && ((weight == "0" && isValid) || (isValid && hasNoConflict)) {
```

Comment:
you can check the valid condition first. if false, return not available. (our controller will set the valid condition as false if the annotation value itself is valid).

then if the weight != 0, check the hasNoConflict condition.
Otherwise, skip the check.